### PR TITLE
util/profile: remove github.com/lonng/pprof replacement

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

#12986 introduced a new dependency `github.com/google/pprof` which was replaced by a customized version `github.com/lonng/pprof` since we need to use several internal packages from the module. However, such change was rejected upstream (google/pprof#496).

This PR re-implements those internal features as suggested by google/pprof's maintainers, and removes the replacement directive (and thus prevents this `replace` from affecting all reverse-dependencies of `tidb`).

### What is changed and how it works?

1. Copied the `"github.com/google/pprof/internal/measurement".Percent` function into this repository, but modified the format slightly so that `strings.TrimSpace` is no longer needed.
2. Rewritten the flamegraph table construction `(*Collector).profileToDatums` function from scratch. 
3. Added a test case to verify the output of `(*Collector).profileToDatums`.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
    * 3.0, 3.1

Release note

### CODE REVIEW

**Comment by lonng** [comment]
@kennytm Nice work. It's better to place the `util/profile/test.pprof` to `util/profile/testdata/test.pprof`.

File: util/profile/profile.go [hunk 1]
```diff
@@ -78,58 +44,19 @@ func (c *Collector) profileReaderToDatums(f io.Reader) ([][]types.Datum, error)
...
-		CumFormat: config.FormatValue(rootValue),
-		Percent:   strings.TrimSpace(measurement.Percentage(rootValue, config.Total)),
-		Children:  nodes[0:nroots],
+	root := newFlamegraphNode()
+	for _, sample := range p.Sample {
+		root.add(sample)
 	}
 
-	c.collect(rootNode, 0, "", 0, config.Total, len(rootNode.Children) == 0)
+	col := newFlamegraphCollector(p, c.Rows)
```
Comment (lonng) [inline util/profile/profile.go]: There is no need to pass the `c.Rows` to `newFlamegraphCollector` because it is empty at the initial state.

Comment (lonng) [reply util/profile/profile.go]: Furthermore, in the current implementation, we don't need the field `Rows` anymore.

Comment (lonng) [reply util/profile/profile.go]: Furthermore, in the current implementation, we don't need the field `Rows` anymore.

File: util/profile/flamegraph.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		n.cumValue += value
+		if len(locs) == 0 {
+			return
+		}
+
+		// The previous implementation in TiDB identify nodes using location ID,
+		// but `go tool pprof` identify nodes using function ID. Should we follow?
+		loc := locs[len(locs)-1].ID
+		child, ok := n.children[loc]
+		if !ok {
```
Comment (lonng) [inline util/profile/flamegraph.go]: I'm not sure whether is it proper for some sample contains recursive (or cumulate the value to existed children).

Comment (kennytm) [reply util/profile/flamegraph.go]: Recursive call generates the same location at different depths (so we'll see ... → A → A → A → ... in the flamegraph).

**Comment by kennytm** [comment]
PTAL @crazycs520

**Pull request merged**